instance/models/instance.py

```diff
@@ -94,6 +96,8 @@ class Instance(ValidateModelMixin, TimeStampedModel):
     STOPPED = 'stopped'
     TERMINATING = 'terminating'
 
+    ERR_PROVISIONING_FAILED = 'provisioning_failed'
```

Comment:
So, having a single error state is better imho, but the way you implemented isn't what I was suggesting. Happy to discuss the merits of each approach (and @smarnach might chime in on this too), but with this, you still have to define a different error for each status, even if you moved it to a different field. What I was suggesting was:

* When an error happens, leave `server.status` with its current status (for what you're implementing, `PROVISIONING`, to preserve that information
* Add a new field, that could have values like `RUNNING`, `SUCCESS`, `FAILED` - maybe called `status_progress` that would further refine the current status. When an error happen, it would go from `RUNNING` to `FAILED`, and we would know that it's a failed provisioning, without the need for adding new values for each of the current states
* Use it in places like https://github.com/open-craft/opencraft/pull/13/files#diff-772e1d190b3c56d74ec801abe130ba8cR50 where we need to know if the user can start a reprovisioning, by simply testing if it isn't `RUNNING`
* Share this field with the client, and use it to display status icons, and whether we show the reprovisioning icon or not.